cf-update-circl.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python3
+#
+# cf-update-circl.py updates the CIRCL package in src/circl.
+import os
+import sys
+import tempfile
+import subprocess
+
+# Name of the CIRCL package as it will appear in the standard library.
+circl_name = 'circl'
+
+# The git repository form which CIRCL will be cloned.
+circl_repo = 'https://github.com/cloudflare/circl'
+
+# The git branch to checkout.
+circl_branch = 'master'
+
+# The set of CIRCL packages that the standard library depends on.
+circl_pkg_dir = [
```

Comment:
Instead of keeping a list of files/directories to remove, you've switched to keeping a list of files/directories to keep.  My guess is that this will be more work during updates.

Reply:
I think that's true when adding a new CIRCL dependency. However, the motivation for this change is that updating CIRCL --- whether you're adding a new feature or not --- tends to break the build here, sometimes requiring upstream changes. By copying only what we need, we minimize this risk. My bet is that overtime, CIRCL will change more rapidly than cfgo's CIRCL dependencies change. This has been true lately. If this holds true in the future, then I think this will be an improvement.